.cbmc-batch/jobs/aws_add_size_checked/aws_add_size_checked_harness.c

```diff
@@ -2,48 +2,25 @@
 #include <proof_helpers/make_common_data_structures.h>
 
 /**
- * Adds a + b and returns the result in *r. If the result
- * overflows, returns AWS_OP_ERR; otherwise returns AWS_OP_SUCCESS.
- */
-AWS_STATIC_IMPL int __aws_add_u64_checked(uint64_t a, uint64_t b, uint64_t *r) {
-    uint64_t x = a + b;
-    *r = x;
-    if (x < a) {
-        return aws_raise_error(AWS_ERROR_OVERFLOW_DETECTED);
-    }
-    return AWS_OP_SUCCESS;
-}
-
-/**
- * Adds a + b and returns the result in *r. If the result
- * overflows, returns AWS_OP_ERR; otherwise returns AWS_OP_SUCCESS.
- */
-AWS_STATIC_IMPL int __aws_add_u32_checked(uint32_t a, uint32_t b, uint32_t *r) {
-    uint32_t x = a + b;
-    *r = x;
-    if (x < a) {
-        return aws_raise_error(AWS_ERROR_OVERFLOW_DETECTED);
-    }
-    return AWS_OP_SUCCESS;
-}
-
-/**
+ * Coverage: 1.00 (29 lines out of 29 statically-reachable lines in 5 functions reached)
  * Runtime: 0m2.025s
+ *
+ * Assumptions:
+ *     - For any given inputs, r will never overflow, if aws_add_u32_checked or
+ *       aws_add_u64_checked functions return AWS_OP_SUCCESS.
  */
 void aws_add_size_checked_harness() {
```

Comment:
I'm confused about what this harness does.  It is called add_size_checked, but then it calls add_u64_checked

Reply:
There are 2 function calls in `aws_add_size_checked` as you can see [here](https://github.com/awslabs/aws-c-common/blob/ac02e1728d740bb9106b6ea727cd3378f8ea438a/include/aws/common/math.h#L148), which depend on the architecture. The same is valid for all implementations in `math.h`.

So, I directly call both implementations, instead of specifying `--64` or `--32` for CBMC.